core/crypto/poseidon_hash.go

```diff
@@ -0,0 +1,177 @@
+package crypto
+
+import (
+	"math/big"
+
+	"github.com/NethermindEth/juno/core/felt"
+)
+
+const (
+	fullRounds    = 8
+	partialRounds = 83
+)
+
+func addRoundKeys(state []felt.Felt, round int) {
+	state[0].Add(&state[0], &roundKeys[round][0])
+	state[1].Add(&state[1], &roundKeys[round][1])
+	state[2].Add(&state[2], &roundKeys[round][2])
```

Comment:
```suggestion
	for i := 0; i < 3; i++ {
		state[i].Add(&state[i], &roundKeys[round][i])
	}
```

Reply:
loops are essentially branches an incur a performance penalty. See "loop unrolling" and https://dev.to/jobinrjohnson/branchless-programming-does-it-really-matter-20j4

Reply:
Thanks for the link